### PR TITLE
SNOW-832825: Fix async documentation

### PR DESCRIPTION
**Pull request by sfc-gh-pbulawa**
### Description
Fix the async example in documentation.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -797,28 +797,17 @@ and before retrieving the results.
...
-				} else if status == QueryFailed {
-					// do something
-				}
-				return nil
-			})
+			// Enable asynchronous mode
+			ctx := sf.WithAsyncMode(context.Background())
+
+			// Run the query with asynchronous context
+			rows, _ := db.QueryContext(ctx, "select 1")
```
Comment (sfc-gh-dprzybysz) [inline doc.go:804]: is it a good practice to ignore the second parameter? is it an error? I understand that if the passed query is wrong then we receive quick answer that sth is wrong?

File: doc.go [hunk 2]
```diff
@@ -797,28 +797,17 @@ and before retrieving the results.
...
+
+			// do something as the workflow continues whereas the query is computing in the background
+			...
+
+			// Get the data when you are ready to handle it
+			cols, err := rows.Columns()
```
Comment (sfc-gh-pfus) [inline doc.go:810]: Why fetching columns?

Comment (sfc-gh-pbulawa) [reply doc.go]: I took it from the `cmd/async.go` example. Replaced it with `.Scan()`.

File: doc.go (hunk 1 above)
Comment (sfc-gh-pbulawa) [reply doc.go]: It is a common practice to skip the errors - in this case I wanted to keep the example, however you are right. This has been addressed.

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Review by sfc-gh-pfus — APPROVED** [review]

**Pull request merged**